### PR TITLE
feat(presto-client): allow basic authentication

### LINKED ISSUE #15: Allow passing any custom headers to the request
This is in case someone has any kind of custom authorization in place, or that requires any specific header.

### LINKED ISSUE #16: Allow authentication

### PR DESCRIPTION
**Pull request by alonsovb**
## Changes

Resolves #15 
Resolves #16 

- Add basic and custom authentication params. The custom authorization param takes precedence over the `basicAuthentication` object.
- Add ability to send custom headers on all requests.

### CODE REVIEW

File: presto-client/src/client.ts [hunk 1]
```diff
@@ -197,7 +226,7 @@ export class PrestoClient {
     const data = []
 
     do {
-      const response = await this.request({ method: 'GET', url: nextUri })
+      const response = await this.request({ headers, method: 'GET', url: nextUri })
```
Comment (alonsovb) [inline presto-client/src/client.ts:229]: I'm sending ALL custom + X-Presto headers here on sub-sequent requests. Let me know if we should limit it. I haven't found any issues by doing so.
But as per @yhwang monitoring of `presto-cli`, it seems the CLI only sends X-Presto-User on these requests.

File: presto-client/README.md [hunk 2]
```diff
@@ -46,6 +46,9 @@ The Presto client can be configured with the following parameters:
 - `schema`: The default schema to use for queries. (Default: `undefined`)
 - `source`: The name of the source you want to use for reporting purposes (Default: `presto-js-client`)
 - `timezone`: The timezone to use for queries. (Default: `undefined`)
+- `authorizationToken`: The value to send as-is in the Authorization header. (Default: `undefined`)
```
Comment (jorgeramirezamora) [inline presto-client/README.md]: Should we use the same order that we have in the client's constructor?

File: presto-client/src/client.ts [hunk 3]
```diff
@@ -46,7 +63,19 @@ export class PrestoClient {
       this.headers['X-Presto-Time-Zone'] = this.timezone
     }
 
-    // TODO: Set up auth
+    if (authorizationToken) {
+      this.headers['Authorization'] = authorizationToken
+    } else if (basicAuthentication) {
+      // Note this is only available for Node.js
+      this.headers['Authorization'] = `Bearer ${Buffer.from(
```
Comment (yhwang) [inline presto-client/src/client.ts]: I believe for basic auth, the auth scheme is `Basic` and `Bearer` is for auth token

Comment (alonsovb) [reply presto-client/src/client.ts]: You're right, I fixed it now

Comment (yhwang) [reply presto-client/src/client.ts]: So the question now is for the auth token, Do you want users to pass in `Bearer <token>` or just `<token>`?
For the latter case, you need to add `Bearer` as the auth scheme in line 67

Comment (alonsovb) [reply presto-client/src/client.ts]: True. What do you think makes more sense?

Comment (yhwang) [reply presto-client/src/client.ts]: I vote for adding the `Bearer` auth scheme in the `client.ts` and users only pass in the auth token

Comment (alonsovb) [reply presto-client/src/client.ts]: @yhwang I just sent a commit with the change you mentioned 👍

**Review by yhwang — APPROVED** [review]
LGTM! nice!

**Review by jorgeramirezamora — APPROVED** [review]
🚀

File: presto-client/README.md (hunk 2 above)
Comment (alonsovb) [reply presto-client/README.md]: in this case I think I rather have params in order of relevance for anyone using the client 🤔

**Pull request merged**